src/cedict-service/indexed-db-store.js

```diff
@@ -0,0 +1,228 @@
+   *
+   * @param {any} key - A key that specifies which records to retrieve.
+   * @param {object} options - Additional configuration parameters:
+   * @param {string} options.index - If the key provided as a first argument is for a secondary index
+   *        then this field must contain a name of a secondary index to use. If this field is not specified,
+   *        then records will be retrieved using a primary index.
+   * @returns {Promise<object[]>|Promise<Error>} A promise that is resolved with an array of records if records
+   *          exist in a store or with an empty array if not. A promise rejection is returned if operation failed.
+   */
+  get (key, options) {
+    return this.getEntries([key], options).then((result) => result[key])
+  }
+
+  /**
+   * Retrieves all records from the store for one or several keys. Options object is implementation specific.
+   *
+   * @param {any|any[]} keys - A key or an array of keys that specifies which records to retrieve.
+   * @param {object} options - Additional configuration parameters:
+   * @param {string} options.index - If the key provided as a first argument is for a secondary index
+   *        then this field must contain a name of a secondary index to use. If this field is not specified,
+   *        then records will be retrieved using a primary index.
+   * @returns {Promise<{key: object[] }>|Promise<Error>} A promise that is resolved with an object. If contains keys
+   *          as its properties and values are arrays of records.
+   *          A promise rejection is returned if operation failed.
+   */
+  getEntries (keys, { index = '' } = {}) {
+    return new Promise((resolve, reject) => {
+      if (!this._db) reject(new Error('Database object is missing'))
+      if (!keys) keys = reject(new Error('No keys provided'))
+      if (!Array.isArray(keys)) keys = [keys]
+      const transaction = this._db.transaction(this._configuration.name, IndexedDbStore.accessModes.READ)
+      const store = transaction.objectStore(this._configuration.name)
+      // Create an object with keys as its props
+      let result = keys.reduce((accumulator, key) => { accumulator[key] = []; return accumulator }, {}) // eslint-disable-line prefer-const
+      /*
+      The order of request execution is guaranteed in IndexedDB.
+      This means that when the last request is completed all previous requests are done too.
+       */
+      for (let i = 0; i < keys.length; i++) {
+        const key = keys[i]
+        let getRequest
+        if (index) {
+          // Use index to retrieve a record
+          const dbIndex = store.index(index)
+          getRequest = dbIndex.getAll(IDBKeyRange.only(key))
+        } else {
+          getRequest = store.getAll(IDBKeyRange.only(key))
+        }
+        getRequest.onsuccess = () => {
+          result[key] = getRequest.result
+          if (i === keys.length - 1) {
+            // A last request has been completed
+            resolve(result)
```

Comment:
same question as above about where the promise is rejected

Reply:
Same as above